# Importation of libraries ,Loading and cleaning of the datasets

In [25]:
import pandas as pd
import numpy as np

from sklearn. pipeline import Pipeline
from sklearn. impute import SimpleImputer
from sklearn. preprocessing import StandardScaler
from sklearn. ensemble import RandomForestClassifier
from sklearn. metrics import classification_report, confusion_matrix, roc_auc_score

In [26]:
raw_df = pd.read_csv("../data/raw_data.csv")
test_df = pd.read_csv("../data/test.csv")
train_df = pd.read_csv("../data/train.csv")

### cleaning the raw_df 

In [27]:
raw_df.head()

Engine rpm  Lub oil pressure  Fuel pressure  Coolant pressure  \
0         700          2.493592      11.790927          3.178981   
1         876          2.941606      16.193866          2.464504   
2         520          2.961746       6.553147          1.064347   
3         473          3.707835      19.510172          3.727455   
4         619          5.672919      15.738871          2.052251   

   lub oil temp  Coolant temp  Engine Condition  
0     84.144163     81.632187                 1  
1     77.640934     82.445724                 0  
2     77.752266     79.645777                 1  
3     74.129907     71.774629                 1  
4     78.396989     87.000225                 0

### check if there are any missing values in the raw_df.

In [28]:
raw_df.isnull().sum()

Engine rpm          0
Lub oil pressure    0
Fuel pressure       0
Coolant pressure    0
lub oil temp        0
Coolant temp        0
Engine Condition    0
dtype: int64

it is seen that there are no null values in the dataset.

### find if there are any duplicated in the dataset.

In [29]:
raw_df.duplicated().sum()

0

there are no duplicates in the dataset
### get to know more about columns and rows in the dataset


In [30]:
raw_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 19535 entries, 0 to 19534
Data columns (total 7 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   Engine rpm        19535 non-null  int64  
 1   Lub oil pressure  19535 non-null  float64
 2   Fuel pressure     19535 non-null  float64
 3   Coolant pressure  19535 non-null  float64
 4   lub oil temp      19535 non-null  float64
 5   Coolant temp      19535 non-null  float64
 6   Engine Condition  19535 non-null  int64  
dtypes: float64(5), int64(2)
memory usage: 1.0 MB


this shows that there are 7 columns, 19535 rows in the dataset.it also shows that there are no missing values as shown before.there is only 2 data tyes 2intagers and 5 floates

In [31]:
raw_df.describe()

Engine rpm  Lub oil pressure  Fuel pressure  Coolant pressure  \
count  19535.000000      19535.000000   19535.000000      19535.000000   
mean     791.239263          3.303775       6.655615          2.335369   
std      267.611193          1.021643       2.761021          1.036382   
min       61.000000          0.003384       0.003187          0.002483   
25%      593.000000          2.518815       4.916886          1.600466   
50%      746.000000          3.162035       6.201720          2.166883   
75%      934.000000          4.055272       7.744973          2.848840   
max     2239.000000          7.265566      21.138326          7.478505   

       lub oil temp  Coolant temp  Engine Condition  
count  19535.000000  19535.000000      19535.000000  
mean      77.643420     78.427433          0.630509  
std        3.110984      6.206749          0.482679  
min       71.321974     61.673325          0.000000  
25%       75.725990     73.895421          0.000000  
50%       76.817350     78.346662          1.000000  
75%       78.071691     82.915411          1.000000  
max       89.580796    195.527912          1.000000

### cleaning the test_df 

In [32]:
test_df .head()

engine_rpm  lub_oil_pressure  fuel_pressure  coolant_pressure  \
0         634          2.611260      10.453517          2.771041   
1         856          4.043840       5.829366          2.460446   
2         814          4.073486       5.257120          1.866571   
3         379          2.724639       4.712955          1.981593   
4         868          2.672809       5.273917          1.905387   

   lub_oil_temp  coolant_temp  engine_condition  
0     75.863640     79.245834                 1  
1     73.860133     71.047078                 1  
2     81.840070     76.715591                 0  
3     77.092260     76.106727                 1  
4     78.475478     72.088361                 0

In [33]:
test_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3907 entries, 0 to 3906
Data columns (total 7 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   engine_rpm        3907 non-null   int64  
 1   lub_oil_pressure  3907 non-null   float64
 2   fuel_pressure     3907 non-null   float64
 3   coolant_pressure  3907 non-null   float64
 4   lub_oil_temp      3907 non-null   float64
 5   coolant_temp      3907 non-null   float64
 6   engine_condition  3907 non-null   int64  
dtypes: float64(5), int64(2)
memory usage: 213.8 KB


from .info() it is seen that there are no missing values,there are 7 columns, 3907 rows and there is only two datatypes 2 intagers and 5 floats.

In [34]:
test_df.duplicated().sum()

0

it is sen that there are no duplicates in the dataset.
### cleaning the train dataset

In [35]:
train_df.head()

engine_rpm  lub_oil_pressure  fuel_pressure  coolant_pressure  \
0         496          4.706240       5.249898          3.803385   
1         836          3.960027       4.928546          3.705357   
2         623          3.055244       9.494612          2.084725   
3         675          2.785797       5.463947          1.649817   
4        1412          2.816022       4.001410          1.921522   

   lub_oil_temp  coolant_temp  engine_condition  
0     75.033607     70.542095                 0  
1     74.395016     84.098307                 1  
2     76.865701     72.927688                 0  
3     75.397279     82.907453                 1  
4     83.246013     71.176927                 1

In [36]:
train_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 15628 entries, 0 to 15627
Data columns (total 7 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   engine_rpm        15628 non-null  int64  
 1   lub_oil_pressure  15628 non-null  float64
 2   fuel_pressure     15628 non-null  float64
 3   coolant_pressure  15628 non-null  float64
 4   lub_oil_temp      15628 non-null  float64
 5   coolant_temp      15628 non-null  float64
 6   engine_condition  15628 non-null  int64  
dtypes: float64(5), int64(2)
memory usage: 854.8 KB


shows that there are no missing values ,there 7 columns and 15628 rows . there are 2 datatypes 2 intagers and 5 floates

In [37]:
train_df.duplicated().sum()

0

there are no duplicates in the dataset
# analysing raw_data
